content/blog/nats-operator-and-service-accounts.md

```diff
@@ -0,0 +1,337 @@
+	// ...
+}
+```
+
+Usually when running the NATS Server in Docker, it would run as PID 1
+inside of the container but in this case since PID namespace sharing
+has been enabled it is logging `6` instead below:
+
+```
+kubectl -n nats-io logs example-nats-1 nats | head
+[6] [INF] Starting nats-server version 1.2.0
+[6] [INF] Git commit [6608e9a]
+[6] [INF] Starting http monitor on 0.0.0.0:8222
+[6] [INF] Listening for client connections on 0.0.0.0:4222
+[6] [INF] Server is ready
+[6] [INF] Listening for route connections on 0.0.0.0:6222
+```
+
+Using `kubectl` we can confirm see there are `2/2` containers running
+for each one of the pods as well (nats server + reloader sidecar).
+
+```
+kubectl -n nats-io get pods -l nats_cluster=example-nats
+NAME                    READY     STATUS    RESTARTS   AGE
+example-nats-1          2/2       Running   0          4h
+example-nats-2          2/2       Running   0          4h
+example-nats-3          2/2       Running   0          4h
+```
+
+Whenever the operator applies updates to the shared secret with the
+configuration of the roles, the reloader sidecar will detect the
+change in the mounted shared secret volume and send the HUP signal to the
+NATS server so that eventually all the servers have the same configuration.
+
+### Conclusions
+
+Although these couple of features are currently being labeled as
+alpha, they are close to being the right way on how to handle these
+issues natively in Kubernetes in a relatively simple way without
+making further changes to the NATS Server.
+
+Another interesting feature that is also worth looking at from latest
+Kubernetes v1.11 release, is the [Service Account Volume Projections](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#service-account-volume-projection)(proposal doc [here](https://github.com/mikedanese/community/blob/2bf41bd80a9a50b544731c74c7d956c041ec71eb/contributors/design-proposals/storage/svcacct-token-volume-source.md)),
+that improves the usability of mounting bound tokens for users.
+
+With this feature a Pod could be defined to use a certain
+service account and then as part of its spec, define the volume which
+will contain the token, making it unecessary to have an extra
+`Secret`, plus the benefit of having a single identity set for each
+one of the Pod workloads (instead of a single shared secret for all
+the NATS client pods as in the current implementation).
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nats-client-pod
+spec:
+  serviceAccountName: "nats-user"
+  volumes:
+    - name: "token"
+      projected:
+        sources:
+        - serviceAccountToken:
+            audience: "nats://example-nats.nats-io.svc"
+            path: "token"
+```
+
+Currently the NATS server does not natively support JWT based auth,
+although it has been requested a few times already, and this feature
+looks like a promising extension point that could be used together
+alongside the `TokenReview API`.
```

Comment:
Although requested previously, the current release of the NATS server does not natively support JWT based authorization, but his feature looks like a promising extension point that could be used together alongside the `TokenReview API`.